math/ncr_modulo_p.cpp

```diff
@@ -101,50 +125,66 @@ class NCRModuloP {
             return 1;
         }
         // fac is a global array with fac[r] = (r! % p)
-        int64_t denominator = modInverse(fac[r], p);
-        if (denominator < 0) {  // modular inverse doesn't exist
-            return -1;
-        }
-        denominator = (denominator * modInverse(fac[n - r], p)) % p;
-        if (denominator < 0) {  // modular inverse doesn't exist
+        const auto denominator = (fac[r] * fac[n - r]) % p;
+        const auto denominator_inv = utils::modInverse(denominator, p);
+        if (denominator_inv < 0) {  // modular inverse doesn't exist
             return -1;
         }
-        return (fac[n] * denominator) % p;
+        return (fac[n] * denominator_inv) % p;
     }
 };
 }  // namespace ncr_modulo_p
 }  // namespace math
 
-/**
- * @brief Test implementations
- * @param ncrObj object which contains the precomputed factorial values and
- * ncr function
- * @returns void
- */
-static void tests(math::ncr_modulo_p::NCRModuloP ncrObj) {
-    // (52323 C 26161) % (1e9 + 7) = 224944353
-    assert(ncrObj.ncr(52323, 26161, 1000000007) == 224944353);
-    // 6 C 2 = 30, 30%5 = 0
-    assert(ncrObj.ncr(6, 2, 5) == 0);
-    // 7C3 = 35, 35 % 29 = 8
-    assert(ncrObj.ncr(7, 3, 29) == 6);
+static void tests() {
```

Comment:
documentation please

Reply:
Documentation should be added so that the function is registered by Doxygen. Yes, test functions _should_ also be detected. 🙂

Reply:
> Documentation should be added so that the function is registered by Doxygen. Yes, test functions _should_ also be detected.

@Panquesito7 what exactly do you mean by that? When I run Doxygen, the output looks like that:
![ncr_mod_p_functions](https://github.com/TheAlgorithms/C-Plus-Plus/assets/65706193/13c0d063-ba76-42ca-8486-886f5012ca25)

Reply:
As you notice, you cannot click on the function and see more detail about it, unlike other *documented* functions.